samza-core/src/main/java/org/apache/samza/storage/TaskSideInputStorageManager.java

```diff
@@ -279,13 +164,12 @@ void writeOffsetFiles() {
   }
 
   /**
-   * Gets the side input SSP offsets for all stores from their local offset files.
+   * Gets the side input SSP offsets for all stores from their local offset files. This method should be executed only
+   * once at class initialization.
```

Comment:
what happens if it is executed after initialization? 
If you really expect this to be only executed once before init, should we have a precondition check and throw exception to explicitly warn about the consequence of this being invoked multiple times?